orc8r/lib/go/service/logcodec.go

```
@@ -24,15 +25,41 @@ import (
 	grpc_proto "google.golang.org/grpc/encoding/proto"
 )
 
+// logModes
+const (
+	DISABLED    string = "0" // Not used
+	FULL               = "1" // Prints all protps
+	HIDEVERBOSE        = "2" // Prints all except verboseProtos
+)
+
+var (
+	// Default list of verbose protos
+	verboseProtos = map[string]bool{
+		"protos.Void":                  true,
+		"protos.ServiceInfo":           true,
+		"protos.HealthStatus":          true,
+		"protos.MetricsContainer":      true,
+		"protos.SubmitMetricsRequest":  true,
+		"protos.SubmitMetricsResponse": true,
+		"grpc.MetricFamilies":          true,
+		"grpc.Void":                    true,
+	}
+)
```

Why don't you make DISABLED/VERBOSE/SHORT a type and make this type a value in your proto logging map:

```
type ProtosLogVerbocity int

const (
	DISABLED = ProtosLogVerbocity(iota)
	VERBOSE
	SHORT
)

var (
	// Default list of verbose protos
	logVerbosity = map[string]ProtosLogVerbocity{
		"protos.Void":                  SHORT,
		"protos.ServiceInfo":           SHORT,
		"protos.HealthStatus":          SHORT,
		"protos.MetricsContainer":      SHORT,
		"protos.SubmitMetricsRequest":  SHORT,
		"protos.SubmitMetricsResponse": SHORT,
		"grpc.MetricFamilies":          SHORT,
		"grpc.Void":                    SHORT,
	}
)

...

type logCodec struct {
	protoCodec encoding.Codec
	logMode    ProtosLogVerbocity
}

```
etc.

---

The reason to have strings instead of ints is for backwards compatibility (due to this [line](https://github.com/magma/magma/pull/10553/files#diff-5130dd3e34b85d89a4f73c2661cb115d910f4a242f620ff9425c8af3e0f00f34L108)). Basically envar `PrintGrpcPayloadEnv` could be 1, true, True or anything else. To avoid issues what i do is, if if `PrintGrpcPayloadEnv` contains any value, then i nable the full log. If that value matches with string "2" then I enable the HIDE feature. 

If i add the ints then, `PrintGrpcPayloadEnv: True` will not b valid anymore (or will requiere special case)

---

I'm strongly in favor of Evgeniy's suggestion here. `logMode` field is too permissive as a `string` type. Another issue with the current implementation here is that it decouples the solution (logMode being string) far away from the issue (PrintGrpcPayloadEnv being multiple different values we support). 

I think a good way to deal with `PrintGrpcPayloadEnv` having different potential values is:
1. `service.go` should by default set `logMode` field to `FULL`, or to `HIDEVERBOSE` if `PrintGrpcPayloadEnv` == "2"
2. We really should throw errors if the input values for an env variable is invalid, or `PrintGrpcPayloadFlag` has an invalid value. If we're too permissive, then we don't really have a well documented public interface.

---

For concern 1, default value is 0 (disabled). And this is current behaviour. We should not enable grpc logging by default due to its verbosity and its possible impact on performance. That is why we keep it separate from the glog level. 

For concern 2, as mentioned in my other comment, this will change the current behaviour. But in case we change that, then it is a good idea to trigger an error in case the value is not right (and default it to 0). 

Our documentation specifies to use 1 to enable grpc logging. But our code says otherwise. So no gap on documentation.